accepted/00000-rbac.md

```diff
@@ -0,0 +1,271 @@
+To mitigate performance impact, a user's access rules can be preloaded into memory and stored in the user's session data during login, enabling faster access control validation.
+
+
+## 4. Management Interface
+
+A UI catalog is proposed for access control administration, offering a clear and user-friendly view of all namespaces along with their descriptions.
+
+Additionally, administrators can manage access through a new API namespace called `access`, which provides methods to:
+
+- Grant or revoke access to a namespace for a user or access group
+- Add or remove a user from an access group
+- Check if a user or group has access to a specific namespace
+- List all available namespaces for a particular feature
+
+
+## 5. Development Resources
+
+After the initial development and mapping of all existing endpoints to the new structure, ongoing effort is required to keep endpoint mappings and the namespace structure up-to-date. To reduce RBAC overhead when implementing new features, the following additional resources are proposed:
+
+- An extensive Wiki guide on adding necessary endpoints with proper namespace organization
+- Stored procedures and scripts to easily grant/revoke user access for development
+- CI checks to detect any unmapped endpoints when introducing new features
+- A failsafe mechanism to prevent access to unmapped endpoints
+- Startup checks in the Uyuni Server to log any misconfigurations within a specific deployment
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- **Increased development complexity and overhead:** Maintaining RBAC will require additional development effort, especially to ensure that all endpoints are properly mapped and categorized. Each new feature will need to be carefully aligned with access policies, adding complexity to the process.
+
+- **Potential for configuration drift**: Over time, there is a risk that the RBAC namespace structure and actual endpoint mappings may drift apart. If new endpoints or features are not consistently mapped or if namespace groupings are mismanaged, this can lead to security gaps or unintended access permissions.
+
+- **Learning curve for administrators and developers**: Introducing a new RBAC model with detailed namespace mapping and access modes may require training for both developers and administrators. Adjusting to this new model and understanding the finer distinctions of access control may initially be challenging.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+- **Granularity of access control:** Should the RBAC model include finer-grained permissions, such as access by specific actions (create, update, delete) within a namespace, or is the proposed "View" vs. "Modify" access mode sufficient for most cases?
+
+- **External integration needs:** Will the RBAC system require compatibility with external authentication or access control systems, such as LDAP, and if so, what integration points are required to support this?
```

Comment:
That I would say is part of authentication. However, we could define a mapping between LDAP user groups and suma internal access groups, and automatically assign users to a group at login time.